### PR TITLE
Allow system tests using Rack::Test to run without Chrome

### PR DESCRIPTION
**Pull request by sinsoku**
### Summary

If you require "action_dispatch/system_test_case", the driven_by method will be executed immediately.

* https://github.com/rails/rails/blob/v6.0.0/actionpack/lib/action_dispatch/system_test_case.rb#L162

Then the preload method is called in SystemTesting::Driver#initialize.

* https://github.com/rails/rails/blob/v6.0.0/actionpack/lib/action_dispatch/system_testing/driver.rb#L13
* https://github.com/rails/rails/blob/v6.0.0/actionpack/lib/action_dispatch/system_testing/browser.rb#L46-L63

Therefore, a "Webdrivers::BrowserNotFound" error occurs in the browser preloading when you run system tests witout Chrome.

This commit avoid the error by lazy configuring the driver.

ref: #37410

### CODE REVIEW

**Comment by y-yagi** [comment]
In my understanding, is this need to fix `Browser#preload` call also?

Currently, `Browser#preload` calls regardless of drivers. If users specify like `driven_by :rack_test`, the `type` is `chrome`. So `::Selenium::WebDriver::Chrome::Service.driver_path` will be call still. 

https://github.com/rails/rails/blob/cb740af5c96ebe0beea01fb8c62e93fb581ad454/actionpack/lib/action_dispatch/system_testing/driver.rb#L13  
https://github.com/rails/rails/blob/cb740af5c96ebe0beea01fb8c62e93fb581ad454/actionpack/lib/action_dispatch/system_testing/browser.rb#L49-L50

**Comment by sinsoku** [comment]
@y-yagi Thank you for your review.
I didn't notice about `Browser#preload` because the reproduction code specified `driven_by(:rack_test, using: :rack_test)`. (refs: [spec/system/foos_spec.rb]( https://github.com/dmolesUC/rack_test_rails_6/blob/8e15721e24a9a0d15e1839ee430ac5f37b4471e1/spec/system/foos_spec.rb#L5))

I fixed it not to preload with :rack_test.

File: actionpack/test/dispatch/system_testing/driver_test.rb [hunk 1]
```diff
@@ -133,4 +133,17 @@ class DriverTest < ActiveSupport::TestCase
   ensure
     ::Selenium::WebDriver::Chrome::Service.driver_path = original_driver_path
   end
+
+  test "does not preload if :rack_test is set" do
+    called = false
```
Comment (y-yagi) [inline actionpack/test/dispatch/system_testing/driver_test.rb]: We can use a helper method to check that method calls or not. Like the following:

```ruby
  test "does not preload if :rack_test is set" do
    assert_not_called_on_instance_of(ActionDispatch::SystemTesting::Browser, :preload) do
      ActionDispatch::SystemTesting::Driver.new(:rack_test, using: :chrome)
    end
  end
```

Comment (sinsoku) [reply actionpack/test/dispatch/system_testing/driver_test.rb]: I updated the code to use the `assert_not_called_on_instance_of` method and  force-pushed it.

**Comment by y-yagi** [comment]
@sinsoku Thanks!

File: actionpack/lib/action_dispatch/system_test_case.rb [hunk 2]
```diff
@@ -157,8 +158,6 @@ def self.driven_by(driver, using: :chrome, screen_size: [1400, 1400], options: {
       self.driver = SystemTesting::Driver.new(driver, **driver_options, &capabilities)
     end
 
-    driven_by :selenium
-
```
Comment (maysam) [inline actionpack/lib/action_dispatch/system_test_case.rb]: I cannot see this in rails:master, not even in the history

**Comment by maysam** [comment]
It is in the master, but was not released in 6.0.2.1

**Comment by tisba** [comment]
This is a really annoying issue that is finally solved. Thank you 💕 

Are there any plan to backport/release this with one of the next patch/minor releases of Rails 6.0.x?

**Comment by y-yagi** [comment]
Sorry for the late response. I backported this to 6-0-stable. So this fix will be included in the next release. Thanks.

**Pull request merged**